packages/core/src/app/customer/CheckoutButtonContainer.tsx

```diff
@@ -1,48 +1,93 @@
-import React, { FunctionComponent } from 'react';
+import classNames from 'classnames';
+import React, { FunctionComponent, useEffect, useState } from 'react';
+
+import { WalletButtonsContainerSkeleton } from '@bigcommerce/checkout/ui';
 
 import { CheckoutContextProps, withCheckout } from '../checkout';
 import { TranslatedString } from '../locale';
 
-import CheckoutButtonListV1 from './CheckoutButtonList';
+import CheckoutButtonListV1, { filterUnsupportedMethodIds } from './CheckoutButtonList';
 
 interface CheckoutButtonsListOnTopProps {
     checkEmbeddedSupport(methodIds: string[]): void;
-
     onUnhandledError(error: Error): void;
 }
 
+const sortMethodIds = (methodIds:string[]):string[] => {
+    const order = ['applepay', 'braintreepaypalcredit', 'braintreepaypal'];
+
+    return methodIds.sort((a, b) => order.indexOf(b) - order.indexOf(a));
+}
+
 const CheckoutButtonContainer: FunctionComponent<CheckoutButtonsListOnTopProps & CheckoutContextProps> = (
     {
         checkEmbeddedSupport,
         checkoutState,
         checkoutService,
         onUnhandledError,
     }) => {
+
     const {
         data: {
             getConfig,
         },
         statuses: {
-            isInitializingCustomer,
+            isInitializedCustomer,
         },
+        errors: {
+            getInitializeCustomerError,
+        }
     } = checkoutState;
     const config = getConfig();
+    const [methodIds, setMethodIds] = useState(filterUnsupportedMethodIds(config?.checkoutSettings.remoteCheckoutProviders ?? []));
+    const isLoading = methodIds.filter(
+        (methodId) => Boolean(getInitializeCustomerError(methodId)) || isInitializedCustomer(methodId)
+    ).length !== methodIds.length;
+
+    useEffect(() => {
```

Comment:
I think you can lift up most up of the filtering / mapping to the `withCheckout` mapper function. That way you don't need to use `useEffect` and `setState` inside `CheckoutButtonContainer` component.

```tsx
export default withCheckout(({
    data: {
        getConfig,
    },
    statuses: {
        isInitializedCustomer,
    },
    errors: {
        getInitializeCustomerError,
    }
},
{
    checkEmbeddedSupport,
    onUnhandledError,
}) => {
    const availableMethodIds = filterUnsupportedMethodIds(getConfig()?.checkoutSettings.remoteCheckoutProviders);
    const methodIds = availableMethodIds.filter(isInitializedCustomer);
    const isLoading = availableMethodIds.filter(
        (methodId) => getInitializeCustomerError(methodId) || isInitializedCustomer(methodId)
    ).length !== methodIds.length;

    return {
        checkEmbeddedSupport,
        checkoutService,
        isLoading,
        onUnhandledError,
        methodIds,
    };
})(CheckoutButtonContainer);
```

Reply:
Thanks DC, I have updated the mapper function like you suggested.

Isn't this because we don't want more state management when we already have it in the SDK?

Reply:
Thanks @bc-peng. Yes, generally speaking, we want to [avoid deriving states from props](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state). In this case, the parent component (checkout HOC) controls the initialised / loading state of the buttons. So its child component (button container) can simply accept the data as props. Otherwise, if the button container also calls `setState`, there are be two places managing the initialised / loading state of the buttons.